File: 3/python/main.py
ASCII_MINUS = 96

def main() -> None:
    data = input().split()
    x = data[0]
    y = int(data[1])

    x = ord(x) - ASCII_MINUS
    if 1 not in (x, y) and 8 not in (x, y):
        print(8)

    elif x in (1,8) and y in(1,8):
        print(3)

    else: 
        print(5)

if __name__ == "__main__":
    main()

File: 3/php/main.php
<?php
    // start code with comand 'php main.php'
    define("ASCII_MINUS", 96);

    function main(){
        [$x , $y] = explode(' ',readline());
        $x = ord($x) - ASCII_MINUS;

        if (!in_array(1, [$x, $y]) && !in_array(8, [$x, $y]))
            echo 8, PHP_EOL;
        else if(in_array($x, [1,8]) && in_array($y, [1,8]))
            echo 3, PHP_EOL;
        else
            echo 5, PHP_EOL;
    }
    main();

